app/src/main/java/com/yalta/viewmodel/DriverRoutesViewModel.kt

```diff
@@ -0,0 +1,93 @@
+package com.yalta.viewmodel
+
+import androidx.databinding.library.baseAdapters.BR
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.yalta.R
+import com.yalta.services.RoutesService
+import com.yalta.utils.UniversalRecyclerItem
+import common.Route
+import common.RoutePoint
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import org.joda.time.DateTime
+import org.joda.time.DateTimeZone
+import org.joda.time.format.DateTimeFormat
+import java.util.*
+import javax.inject.Inject
+
+class DriverRoutesViewModel @Inject constructor(
+    private val routesService: RoutesService,
+    private val dispatcher: CoroutineDispatcher
+) : ViewModel() {
+    var routes: List<Route>? = emptyList()
+    val routesNames = MutableLiveData<List<String>>()
+    var selectedRoute: Int? = null
+    val selectedRoutePoints = MutableLiveData<List<UniversalRecyclerItem>>()
+    val fromDate = MutableLiveData<DateTime>()
+    val toDate = MutableLiveData<DateTime>()
+    val showDatesError = MutableLiveData(false)
+
+    fun getSomeRoutes() {
+        var from = fromDate.value
+        var to = toDate.value
+        val timezone = DateTimeZone.forTimeZone(TimeZone.getTimeZone("Europe/Moscow"))
+        val now = DateTime.now()
+
+        from = from ?: DateTime(now.year, now.monthOfYear, now.dayOfMonth, 0, 0, timezone)
+        to = to ?: DateTime(now.year, now.monthOfYear, now.dayOfMonth, 23, 59, timezone)
+
+        if (datesAreOk(from, to)) {
+            getSelectedRoutes(from, to)
+        } else {
+            showDatesError(true)
+        }
+    }
+
+    private fun datesAreOk(from: DateTime?, to: DateTime?): Boolean =
+        from != null && to != null && from <= to
+
+    private fun getSelectedRoutes(from: DateTime, to: DateTime) = viewModelScope.launch(dispatcher) {
+        routes = routesService.getRoutes(from.toString(), to.toString())
+        if (routes != null) {
+            setRoutes()
+        }
+    }
+
+    private fun setRoutes() = viewModelScope.launch(Dispatchers.Main) {
+        routesNames.value = routes?.map { route ->
+            "ID: ${route.id} Date: ${route.routeDate.toString(DateTimeFormat.forPattern("dd-MM-yy HH:mm"))}"
+        }
+        selectedRouteChanged(null)
```

Comment:
Why drop selection? maybe select first?

Reply:
It's needed in case we've received empty list
In case it' not empty the first element is autoselected